lib/bamboo/adapters/smtp_adapter.ex

```diff
@@ -71,11 +71,18 @@ defmodule Bamboo.SMTPAdapter do
       config
       |> to_gen_smtp_server_config
 
-    email
-    |> Bamboo.Mailer.normalize_addresses()
-    |> to_gen_smtp_message
-    |> config[:transport].send_blocking(gen_smtp_config)
-    |> handle_response
+    response =
+      try do
+        email
+        |> Bamboo.Mailer.normalize_addresses()
+        |> to_gen_smtp_message
+        |> config[:transport].send_blocking(gen_smtp_config)
+      catch
+        e ->
+          handle_response({:error, e})
```

Comment:
```suggestion
          raise SMTPError, {:not_specified, detail}
```

I think there is no need to add a new pattern to handle_response just to raise an error

Reply:
I agree on raising directly the error but I rather keep the pattern, `gen_smtp` behave oddly sometimes and I will not be surprised that instead of throwing an exception they send me an `{:error, detail}` tuple